### DISCUSSION TITLE
Permissions for ncdata and backup location

### DISCUSSION THREAD

**Discussion by ibnzbssrckr** [nextcloud/all-in-one#4434] [Questions] [answered]
I am running nextcloud-aio in a VM on proxmox with the ncdata and backup folders mounted to the host over nfs. The application has always worked correctly but I have never been able to get the backup to work. I believe it is a permissions problem. When I set the permissions on the ncdata folder to 33:0 750 the backup container complains about there not being a .ocdata folder. If I open permissions on the ncdata folder it complains of not being able to write to the backup folder. What are the appropriate ownership/permissions on these folders to get them to work?

**Comment by szaimen** [comment, marked as answer]
Hi, the web server inside AIO runs as userid 33 and the backup container as user root (user id 0). Both need to have access to the files.

**Reply by ibnzbssrckr** [reply]
I have the ownership set appropriately and now cannot get past the backup container saying the .ocdata folder is missing.

`
The .ocdata file is missing in Nextcloud datadir which means it is invalid!
Is the drive where the datadir is located on still mounted?
touch: /nextcloud_aio_volumes/nextcloud_aio_nextcloud_data/skip.update: Permission denied
chmod: /nextcloud_aio_volumes/nextcloud_aio_nextcloud_data/skip.update: Permission denied
`

**Reply by szaimen** [reply]
Did you maybe forbid root access on your nfs share?

**Reply by ibnzbssrckr** [reply]
Thank you so much! For future reference I had to set the zfs nfs export like this with the option `no_root_squash`

`zfs set sharenfs='no_root_squash,rw=@xxx.xxx.xxx.xxx'` on both the exports

I do wonder about the security implications of this, even though it is completely behind a firewall. I may need to rethink the way I am doing things but I am so happy to have backups working.
